src/process-events.c

```diff
@@ -146,6 +146,13 @@ int kprobe__do_mount(struct pt_regs *ctx)
 
 static __always_inline void push_telemetry_event(struct pt_regs *ctx, ptelemetry_event_t ev)
 {
+    // check mm field of task_struct
+    // skip pushing kernel process events as they have an mm field of NULL
+    void *ts = (void *)bpf_get_current_task();
+    void *ptr = NULL;
+    read_value(ts, CRC_TASK_STRUCT_MM, &ptr, sizeof(ptr));
+    if (!ptr) return;
```

Comment:
I'm not sure if this is the appropriate place to implement this filtering. My reasoning was that it would be better here than duplicating these lines in several places.

Reply:
I think that having it here is a bit of an overkill. Doing this check on every push is probably not necessary since some syscalls send multiple events but really we just need to check it once per syscall (or maybe even only on exit, unsure if the other syscalls would be triggered at all by kernel processes). 

It might fit better either explicitly in the exit syscalls for kernels > 4.8, or in `GET_OFFSETS_4_8` since we already get the current task there.